src/base/preferences.cpp

```diff
@@ -561,6 +561,26 @@ void Preferences::setWebUiHttpsKey(const QByteArray &data)
     setValue("Preferences/WebUI/HTTPS/Key", data);
 }
 
+bool Preferences::isAltWebUiEnabled() const
+{
+    return value("Preferences/WebUI/AlternativeUIEnabled", false).toBool();
+}
+
+void Preferences::setAltWebUiEnabled(bool enabled)
+{
+    setValue("Preferences/WebUI/AlternativeUIEnabled", enabled);
+}
+
+QString Preferences::getWebUiRootFolder() const
+{
+    return value("Preferences/WebUI/RootFolder").toString().trimmed();
+}
+
+void Preferences::setWebUiRootFolder(const QString &path)
+{
+    setValue("Preferences/WebUI/RootFolder", path.trimmed());
```

Comment:
A path may end with a space. Why do you reject that?

Reply:
Technically, Yes. But as soon as to have a space at the beginning or end of a phrase is grammatically incorrect, more likely it's just a typo.
Similar corrections are found everywhere in our code, so I just did it the same way. It doesn't matter much for me, so if you insist, I'll change it.

Reply:
I think that trimming file paths is counter-intuitive and we shall not do this.

Reply:
Ok. Let users to be careful.